src/coreComponents/constitutive/fluid/ReactiveBrineFluid.hpp

```diff
@@ -0,0 +1,358 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file ReactiveBrineFluid.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_FLUID_REACTIVEBRINEFLUID_HPP_
+#define GEOSX_CONSTITUTIVE_FLUID_REACTIVEBRINEFLUID_HPP_
+
+#include "codingUtilities/EnumStrings.hpp"
+#include "constitutive/fluid/ReactiveMultiFluid.hpp"
+#include "constitutive/fluid/MultiFluidUtils.hpp"
+#include "constitutive/fluid/PhaseModel.hpp"
+#include "constitutive/fluid/PVTFunctions/BrineEnthalpy.hpp"
+#include "constitutive/fluid/PVTFunctions/NoOpPVTFunction.hpp"
+#include "constitutive/fluid/PVTFunctions/PhillipsBrineDensity.hpp"
+#include "constitutive/fluid/PVTFunctions/PhillipsBrineViscosity.hpp"
+
+#include <memory>
+
+namespace geosx
+{
+
+namespace constitutive
+{
+
+template< typename PHASE >
+class ReactiveBrineFluid : public ReactiveMultiFluid
+{
+public:
+
+  using exec_policy = parallelDevicePolicy<>;
```

Comment:
Maybe you could just have `exec_policy = serialPolicy;` in this PR, since it seems really close to be mergeable for the CPU case, and worry about the GPU case in a later PR. For instance with `serialPolicy`, you would not have to worry about the call to
```
BlasLapackLA::solveLinearSystem( matrix, rhs, solution );
```
and you could just merge the PR as is.